src/Core/StreamableBody.php

```diff
@@ -43,24 +43,23 @@ public function getContentAsString(): string
     {
         $resource = $this->getContentAsResource();
 
-        return \stream_get_contents($resource);
+        try {
+            return \stream_get_contents($resource);
+        } finally {
+            \fclose($resource);
+        }
     }
 
     public function getContentAsResource()
     {
         $resource = \fopen('php://temp', 'rw+');
```

Comment:
shouldn't it be closed if method failed?

Reply:
Yeah, Good